scipy/optimize/tests/test_minpack.py

```diff
@@ -517,6 +517,131 @@ def test_maxfev_and_bounds(self):
         assert_allclose(popt1, 2, atol=1e-14)
         assert_allclose(popt2, 2, atol=1e-14)
 
+    def test_curvefit_simplecovariance_nojacobian(self):
```

Comment:
I suggest to optimize new tests in a simple manner. Just loop over options for the Jacobian ``for jac in [None, jacp]``. So we need only 2 test functions.